### ISSUE TITLE
How to stop heartbeat goroutine for lost lock?

### ISSUE THREAD

**Issue by tomaszhrem** [cirello-io/dynamolock#560]
Hello,
is there any way to stop sending heartbeats when process which aquire lock is interrupted?
I tried to cancel context which was passed to `AcquireLock` and also finishing goroutine which created `DynamoLock` and later acquired lock, but in both cases some background process is still sending heartbeats.

My use case is that task which hold the lock is finished but release is not executed due to its termination and I need to wait whole TTL to get be able to lock again. As task I understand e.g. handling http request which is indepenededn from `main()` process. main() is still running.,

**Comment by ucirello** [comment]
hi @tomaszhrem -- thanks for opening this issue. I am not sure I follow what you are trying to do. Could you please share a code example with the behavior you're describing so that I can try and reproduce the problem (leaking heartbeats)?

**Comment by tomaszhrem** [comment]
Let me rephrase it - apart of unlocking is there any way to stop sending heartbeats for lock in case of stoping goroutine which created that lock?
In my use case worker which acquire the lock was interrupted but `defer` with unlock was not called (still looking fo cause of that).

I was hoping that maybe canceling context used to create lock could or killing process which created client would help but none of those work. Here is an example for this )context cancel, and stopping client and lock goroutine):
```
        dynamoDBClient := // my config
	tableName := "my_lock_table"
	leaseDuration := 20 * time.Second
	heartbeatPeriod := 1 * time.Second
	queryCtx, cancel := context.WithTimeout(context.TODO(), 5*time.Second)
	go func() {
		defer cancel()
		client, err := dynamolock.New(
			dynamoDBClient,
			tableName,
			dynamolock.WithLeaseDuration(leaseDuration),
			dynamolock.WithHeartbeatPeriod(heartbeatPeriod),
			dynamolock.WithOwnerName("OwnerName"),
		)
		if err != nil {
			fmt.Println("could not instantiate locking client %w", err)
			return
		}

		lockOptions := []dynamolock.AcquireLockOption{
			dynamolock.WithDeleteLockOnRelease(),
			dynamolock.WithData([]byte("MyData")),
			dynamolock.WithAdditionalAttributes(map[string]dynamoTypes.AttributeValue{
				"TTL": &dynamoTypes.AttributeValueMemberN{Value: fmt.Sprint(time.Now().Add(5 * time.Minute).Unix())},
			})}
		//}

		lock, err := client.AcquireLockWithContext(
			queryCtx,
			"LOCK_KEY",
			lockOptions...,
		)
		fmt.Println("IS LOCKED:", lock.IsExpired())
	}()

	time.Sleep(5 * time.Second)
	fmt.Println(queryCtx.Err())
```

**Comment by ucirello** [comment]
hi @tomaszhrem -- in the past, when I faced similar situation, my line of action was to not use automatic heartbeats at all. I recommend you to do the same ( https://pkg.go.dev/cirello.io/dynamolock/v2#DisableHeartbeat ) . Manual Heartbeats ( https://pkg.go.dev/cirello.io/dynamolock/v2#Client.SendHeartbeatWithContext ) is the best tool for the job here, so that you can control when and why you call them.
